src/embit/bip47.py

```diff
@@ -0,0 +1,258 @@
+import hashlib
+import hmac
+import sys
+
+from binascii import hexlify, unhexlify
+from io import BytesIO
+from typing import Tuple
+
+from . import base58, ec, script
+from .base import EmbitError
+from .bip32 import HDKey
+from .networks import NETWORKS
+from .script import OPCODES
+from .transaction import Transaction
+if sys.implementation.name == "micropython":
+    import secp256k1
+else:
+    from .util import secp256k1
+
+
+"""
+    BIP-47: https://github.com/bitcoin/bips/blob/master/bip-0047.mediawiki
+"""
+
+class BIP47Exception(Exception):
+    pass
+
+
+def get_payment_code(root: HDKey, coin: int = 0, account: int = 0) -> str:
+    """
+        Generates the recipient's BIP-47 shareable payment code (version 1)
+        for the input root private key.
+    """
+    bip47_child = root.derive(f"m/47'/{coin}'/{account}'")
```

Comment:
f-strings are not supported by micropython

Reply:
Gackk! My beloved f-strings! Fixed.